hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -255,3 +260,35 @@ pluginResponse :: Monad m => ExceptT String m a -> m (Either ResponseError a)
 pluginResponse =
   fmap (first (\msg -> ResponseError InternalError (fromString msg) Nothing))
     . runExceptT
+
+-- ---------------------------------------------------------------------
+
+type TextParser = P.Parsec Void T.Text
+
+-- | Unescape printable escape sequences within double quotes.
+-- This is useful if you have to call 'show' indirectly, and it escapes some characters which you would prefer to
+-- display as is.
+unescape :: T.Text -> T.Text
+unescape input =
+    case P.runParser escapedTextParser "inline" input of
+        Left _     -> input
+        Right strs -> T.pack strs
+
+-- | Parser for a string that contains double quotes. Returns unescaped string.
+escapedTextParser :: TextParser String
+escapedTextParser = do
+    xs <- P.many (P.try stringLiteral)
+    x <- P.manyTill P.anySingle P.eof -- consume characters after the final double quote
+    pure $ concat xs ++ x
+  where
+    stringLiteral :: TextParser String
+    stringLiteral = do
+        before <- P.manyTill P.anySingle (P.char '"') -- include any character before the first double quote
+        inside <- P.manyTill P.charLiteral (P.char '"')
```

Comment:
What about `"hell\"ooo"`? i.e. escaped quotes probably shouldn't stop us

Reply:
Hmm, from the tests it looks like you made this work but I'm not sure why it works...

Reply:
`P.charLiteral` calls `lookAhead` internally. And this is an example from the haddock of `charLiteral`:
```
stringLiteral = char '"' >> manyTill L.charLiteral (char '"')
```